cre/include/state_client.hpp

```diff
@@ -0,0 +1,55 @@
+// Concord
+//
+// Copyright (c) 2018-2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License"). You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+#include <map>
+#include <mutex>
+#include <condition_variable>
+#include <thread>
+#include "bftclient/bft_client.h"
+namespace cre::state {
+struct State {
+  uint64_t block;
+  std::vector<uint8_t> data;
+};
+
+class IStateClient {
```

Comment:
Consider defining interfaces `IStateClient` and `IStateHandler` separately from the implementations.

Reply:
fixed